eurec4a_environment/variables/boundary_layer/mixed_layer_height.py

```diff
@@ -93,3 +91,79 @@ def calc_peakRH_linearize(
     da.attrs["long_name"] = "mixed layer height (from RH peak and linearization)"
     da.attrs["units"] = "m"
     return da
+
+
+def calc_from_gradient(
+    ds, var, threshold, z_min=200, altitude="height", time="sounding"
+):
+    """
+    Find mixed layer height as layer over which x(z+1) - x(z) < threshold
+
+    Inputs:
+        -- ds: Dataset
+        -- var: variable name
+        -- threshold: in units of variable (i.e. g/kg or K). Maximum difference allowed in one vertical step, dz=10m, i.e. threshold_q = 0.4 g/kg,
+            or vertical gradient dq/dz < 0.04 g/kg/m
+
+    Outputs: DataArray containing mixed layer height from gradient method
+
+    Note that function is slow and should be optimized
+    """
+
+    def calculateHmix_var(
```

Comment:
Would it be better to define this outside of the function in case it could be used elsewhere? If you don't want it as a generally visible function you can rename it with an underscore in front (i.e. \_calculateHmix_var)

Reply:
Raising an issue for higher precision tests sounds great -- thank you!

Reply:
And yes that's great with _calculateHmix_var, will do. Thanks for the tip